python_wrap.cxx build failure with VS2022 Preview

When building with VS2022 Preview 4.0 onwards, I get a bunch of errors when compiling python wrapper class which was generated with SWIG 4.0.2 on Windows. This compiles fine with VS2019. Using Python 3.9.7 64bit.

```
[777/779] Building CXX object ui\python\CMakeFiles\py3.dir\3\x86_64\PYTHON_wrap.cxx.obj
  FAILED: ui/python/CMakeFiles/py3.dir/3/x86_64/PYTHON_wrap.cxx.obj 
  C:\PROGRA~1\MIB055~1\2022\Preview\VC\Tools\MSVC\1430~1.307\bin\Hostx64\x64\cl.exe  /nologo /TP -DSWIG_PYTHON_INTERPRETER_NO_DEBUG -D_CRT_SECURE_NO_WARNINGS -Dpy3_EXPORTS -IC:\python\include ...includes removed... /DWIN32 /D_WINDOWS /EHsc /sdl /GR /EHsc /DNOMINMAX /Zi /Ob0 /Od /RTC1 /Od -MDd /W4 /WX /MP /bigobj /WX- /wd5033 -std:c++latest /showIncludes /Foui\python\CMakeFiles\py3.dir\3\x86_64\PYTHON_wrap.cxx.obj /Fdui\python\CMakeFiles\py3.dir\ /FS -c D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx
D:\src\out\build\x64-Debug\cl : Command line warning D9025: overriding '/WX' with '/WX-'
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xmemory(154): error C2039: '_invalid_parameter': is not a member of '`global namespace''
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xmemory(154): error C3861: '_invalid_parameter': identifier not found
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xmemory(164): error C2039: '_invalid_parameter': is not a member of '`global namespace''
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xmemory(164): error C3861: '_invalid_parameter': identifier not found
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xmemory(1203): error C2039: '_invalid_parameter': is not a member of '`global namespace''
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xmemory(1203): error C3861: '_invalid_parameter': identifier not found
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xpolymorphic_allocator.h(146): error C2039: '_invalid_parameter': is not a member of '`global namespace''
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xpolymorphic_allocator.h(146): error C3861: '_invalid_parameter': identifier not found
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xpolymorphic_allocator.h(152): error C2039: '_invalid_parameter': is not a member of '`global namespace''
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\xpolymorphic_allocator.h(152): error C3861: '_invalid_parameter': identifier not found
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\atomic(206): error C2039: '_invalid_parameter': is not a member of '`global namespace''
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\atomic(206): error C3861: '_invalid_parameter': identifier not found
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\atomic(221): error C2039: '_invalid_parameter': is not a member of '`global namespace''
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\atomic(221): error C3861: '_invalid_parameter': identifier not found
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\atomic(237): error C2039: '_invalid_parameter': is not a member of '`global namespace''
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\atomic(237): error C3861: '_invalid_parameter': identifier not found
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\atomic(297): error C2039: '_invalid_parameter': is not a member of '`global namespace''
C:\Program Files\Microsoft Visual Studio\2022\Preview\VC\Tools\MSVC\14.30.30704\include\atomic(297): error C3861: '_invalid_parameter': identifier not found
D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4492): warning C4127: conditional expression is constant
  D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4492): note: consider using 'if constexpr' statement instead
D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4498): warning C4127: conditional expression is constant
  D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4498): note: consider using 'if constexpr' statement instead
D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4525): warning C4127: conditional expression is constant
  D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4525): note: consider using 'if constexpr' statement instead
D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4634): warning C4127: conditional expression is constant
  D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4634): note: consider using 'if constexpr' statement instead
D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4640): warning C4127: conditional expression is constant
  D:\src\ui\python\3\x86_64\PYTHON_wrap.cxx(4640): note: consider using 'if constexpr' statement instead
```

--- a/Lib/python/pyruntime.swg
+++ b/Lib/python/pyruntime.swg
@@ -4,6 +4,13 @@
 # include <math.h>
 #endif
 
+#if defined(_MSC_VER) && _MSC_VER >= 1930
+/* Workaround what seems to be a bug in the Python headers with MSVC 2022.
+ * https://github.com/swig/swig/issues/2090
+ */
+# include <corecrt.h>
+#endif
+
 #if defined(_DEBUG) && defined(SWIG_PYTHON_INTERPRETER_NO_DEBUG)
 /* Use debug wrappers with the Python release dll */
 # undef _DEBUG

